components/v-mapbox-components/v-mapbox-vector-layer.vue

```diff
@@ -93,6 +78,82 @@ export default {
         this.map.getCanvas().style.cursor = ''
       })
     },
+
+    showGraph(layer, event) {
+      this.adjustMap(event, 500)
+
+      const features = this.map.queryRenderedFeatures(event.point)
+      this.$emit('select-locations', {
+        features,
+        geometry: features[0].geometry,
+      })
+    },
+    adjustMap(event, duration) {
+      const { clientWidth } = this.map.getCanvas()
+
+      // prettier-ignore
+      // the timeseries panel is max 600px wide otherwise the half of the screen
+      const visibleMapWidth = clientWidth > 1200
+        ? (clientWidth - 600) * 0.25
+        : (clientWidth / 2) * 0.5
+      const targetLocation = this.map.unproject({
+        x: event.point.x - visibleMapWidth,
+        y: event.point.y,
+      })
+      this.map.panTo(targetLocation, { duration })
+    },
+    zoomTo(layer, event) {
+      // Use zoomto and the click event to zoom in on the map and display the
+      // features of the layertozoomto
+      const zoomLevel = _.get(layer, 'onClick.zoomTo')
+      const duration = 500
+
+      this.map.flyTo({
+        center: event.lngLat,
+        zoom: zoomLevel,
+      })
+
+      setTimeout(() => {
```

Comment:
why  timeout?

Reply:
Or you could to something like this, perhaps in v-mapbox
```
function flyTo(lngLat, zoom) {
 let flyTime = 300
 let promise = new Promise(function(resolve, reject) {
  this.map.flyTo()
  setTimeout(function() {
    resolve()
  }, flyTime)
 })
 return promise
})
```

en dan 
`await this.flyTo()`

Reply:
I think it is a good idea to put it in v-mapbox